packages/analytics-node/src/config.ts

```diff
@@ -0,0 +1,40 @@
+import { NodeOptions, NodeConfig as INodeConfig, Event } from '@amplitude/analytics-types';
+import { Config, Http, MemoryStorage, UUID } from '@amplitude/analytics-core';
+
+export class NodeConfig extends Config implements INodeConfig {
+  constructor(apiKey: string, userId?: string, options?: NodeOptions) {
+    const storageProvider = createEventsStorage();
+    const transportProvider = createTransport();
+
+    super({
+      ...options,
+      apiKey,
+      storageProvider,
+      transportProvider,
+      userId: userId,
+      deviceId: createDeviceId(),
+      sessionId: createSessionId(),
+    });
+  }
+}
+
+export const useNodeConfig = (apiKey: string, userId?: string, overrides?: NodeOptions): INodeConfig => {
+  return new NodeConfig(apiKey, userId, overrides);
+};
+
+export const createEventsStorage = () => {
+  const eventsStorage = new MemoryStorage<Event[]>();
+  return eventsStorage;
+};
```

Comment:
i'd say we can skip createEventStorage and create Transport since they are one liner, then we can just specify default values and spread them in NodeConfig's super call